More SI and test ideal gas equation of state

jedbrown opened the pull request:
I was experimenting with different dimensional analysis packages and this one seems to have the nicest error messages so I implemented my standard test (a representative ingredient in computational fluid dynamics) and thought it might be useful to others. The `diman::si` module was extremely bare-bones so I put in various common quantities for gases/thermodynamics.

I imagine if this test is something you'd like to merge, you may want it converted to use the `gen_tests_for_float!`, but wanted to check before doing that.

Tehforsch commented:
Thanks for this! Looks not only like a good test but also like a nice example. I was planning on "properly" setting up the `si` module (which for now was just a few definitions to make the tests possible) some time soon, but in the meantime having a few more definitions in there cannot hurt.

As for wrapping a `gen_tests_for_float!` macro around this: I am wondering whether there isn't some value in having some tests that are not wrapped in a macro call, but simply normal code, even if that means they don't test every float variant. So while those example-like tests should probably eventually become actual examples, for now I would also be fine with this just being one float type.

So unless you'd prefer otherwise, I would take out the compile_fail_fix for which I merged your other PR (thanks for that, too.) and then merge this.

jedbrown commented:
Cool, I rebased and added a couple more SI quantities that will be useful for anyone working with fluids/to expand the gas example to include inviscid and viscous fluxes.

Tehforsch commented:
thanks!
